HRIS.Services.Tests/Services/EmployeeDocumentServiceUnitTest.cs

```diff
@@ -189,6 +190,60 @@ public async Task SaveEmployeeDocumentOnProfilePass()
 
 
 
+        List<Role> roles = new List<Role> { new Role(EmployeeRoleTestData.RoleDtoEmployee) };
+        _employeeServiceMock.Setup(x => x.GetById(employeeId))
+            .ReturnsAsync(EmployeeTestData.EmployeeDto);
+
+        _unitOfWorkMock
+            .Setup(x => x.EmployeeRole.Get(It.IsAny<Expression<Func<EmployeeRole, bool>>>()))
+            .Returns(EmployeeRoleTestData.EmployeeRolesList.AsQueryable().BuildMock());
+
+        _unitOfWorkMock
+            .Setup(x => x.Role.Get(It.IsAny<Expression<Func<Role, bool>>>()))
+            .Returns(roles.AsQueryable().BuildMock());
+
+        _unitOfWorkMock.Setup(x => x.EmployeeDocument.Add(It.IsAny<EmployeeDocument>()))
+            .ReturnsAsync(EmployeeDocumentTestData.EmployeeDocumentPending);
```

Comment:
Consider encapsulating mock setups into a dedicated method to improve code maintainability code duplication. This will make the test setup more organized and easier to manage.